src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -266,6 +266,47 @@ public bool IsSolutionAvailable
             }
         }
 
+        public bool IsSolutionDPLEnabled
+        {
+            get
+            {
+#if VS15
+                return ThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                    var vsSolution7 = _vsSolution as IVsSolution7;
+
+                    if (vsSolution7 != null && vsSolution7.IsSolutionLoadDeferred())
+                    {
+                        return true;
+                    }
+
+                    return false;
+                });
+#elif VS14
+                // for Dev14 always return false since DPL not exists there.
+                return false;
+#endif
+            }
+        }
+
+        public void EnsureSolutionIsLoaded()
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                IVsSolution4 vsSolution4 = _vsSolution as IVsSolution4;
+
+                if (vsSolution4 != null)
+                {
+                    // ignore result and continue. Since results may be incomplete if user canceled.
+                    vsSolution4.EnsureSolutionIsLoaded((uint)__VSBSLFLAGS.VSBSLFLAGS_None);
```

Comment:
Since most cases users won't cancel this and will continue with the load, but even if user does cancel it, then also idea here is to continue with whatever projects are loaded instead of failing and I feel that's the right approach. There is no benefit of logging this info as well since user explicitly cancel that task. Also I talked to vs folks and xaml team, and this seems to be the right approach.